pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1049,6 +1069,39 @@ func applyUpstreamTLSSettings(env *model.Environment, cluster *apiv2.Cluster, tl
 		}
 	}
 
+	// TODO where to store harcoded strings
+	if env.Mesh.GetEnableAutoMtls().Value {
+		// Per envoy docs
+		// If an endpoint metadata's value under *envoy.transport_socket* does not match any
+		// *TransportSocketMatch*, socket configuration fallbacks to use the *tls_context* or
+		// *transport_socket* specified in this cluster
+		cluster.TransportSocketMatches = []*apiv2.Cluster_TransportSocketMatch{
+			{
+				Name: "enableMtls",
+				Match: &structpb.Struct{
+					Fields: map[string]*structpb.Value{
+						util.AcceptMTLSMetadataLabel: {Kind: &structpb.Value_StringValue{StringValue: "true"}},
+					},
+				},
+				TransportSocket: &core.TransportSocket{
+					Name: "tls",
+					ConfigType: &core.TransportSocket_Config{
+						Config: &structpb.Struct{
+							Fields: map[string]*structpb.Value{},
+						},
+					},
+				},
+			},
+			{
+				Name:  "defaultToPlaintext",
+				Match: &structpb.Struct{},
```

Comment:
nil?

Reply:
This is `raw_buffer`. They has to be matched against the hard-coded built-in Envoy transport socket names.

Check if any constants var are already defined, if not, worth adding one util.EnvoyRawBufferSocket, util.TlsSocketName, (reword if needed).